jenkins/Jenkinsfile.windows

```
@@ -92,214 +92,217 @@ pipeline {
+      stages {
+        stage ('Checkout') {
+          when { expression { params.COMMIT_HASH != '' } }
+          steps {
+            checkout([$class: 'GitSCM',
+              branches: [[name: params.COMMIT_HASH]],
+              extensions: [],
+              userRemoteConfigs: [[credentialsId: 'posit-jenkins', url: GIT_URL]]])
           }
-          //   expression { return SHOULD_REBUILD == true } // Disabling for now (see above)
-          // }
         }
 
-        stages {
-          stage('Build Windows') {
-            agent {
-              docker {
-                image "jenkins/ide:${utils.getDockerTag()}"
-                  registryUrl 'https://263245908434.dkr.ecr.us-east-1.amazonaws.com'
-                  registryCredentialsId 'ecr:us-east-1:aws-build-role'
-                  reuseNode true
-                label "windows"
-                // Set a custom workspace relative to the workspace root (C:\Users\jenikns) to ensure file names
-                // won't be too long
-                customWorkspace "workspace/ide-${IS_PRO ? 'pro' : 'os'}-windows/${env.BRANCH_NAME.replace('/', '-')}"
-              }
-            }
+        stage('Build') {
+          steps {
+            // set requisite environment variables and build rstudio
+            bat "cd package/win32 &&" +
+              "set \"rstudio_version_major=${RSTUDIO_VERSION_MAJOR}\" &&" +
+              "set \"rstudio_version_minor=${RSTUDIO_VERSION_MINOR}\" &&" +
+              "set \"rstudio_version_patch=${RSTUDIO_VERSION_PATCH}\" &&" +
+              "set \"rstudio_version_suffix=${RSTUDIO_VERSION_SUFFIX}\" &&" +
+              "set \"package_os=windows\" &&" +
+              "make-package.bat clean ${FLAVOR.toLowerCase()} &&" +
+              "cd ../.."
+          }
+        }
 
-            stages {
-              stage ('Checkout') {
-                when { expression { params.COMMIT_HASH != '' } }
-                steps {
-                  checkout([$class: 'GitSCM',
-                    branches: [[name: params.COMMIT_HASH]],
-                    extensions: [],
-                    userRemoteConfigs: [[credentialsId: 'posit-jenkins', url: GIT_URL]]])
-                }
-              }
+        stage('Sign Executables') {
+          environment {
+            EXECUTABLES = """package\\win32\\build\\src\\cpp\\session\\rsession-utf8.exe ^
+              package\\win32\\build\\src\\cpp\\session\\rsession.exe ^
+              package\\win32\\build\\src\\cpp\\session\\x86\\rsession.exe ^
+              package\\win32\\build\\src\\cpp\\session\\consoleio\\consoleio.exe ^
+              package\\win32\\build\\src\\cpp\\session\\postback\\rpostback.exe ^
+              package\\win32\\build\\src\\cpp\\diagnostics\\diagnostics.exe ^
+              src\\node\\desktop\\out\\RStudio-win32-x64\\rstudio.exe"""
+          }
+          bat 'curl -o smtools-windows-x64.msi "https://rstudio-buildtools.s3.amazonaws.com/posit-dev/smtools-windows-x64.msi"' 
```

You can use `-O` (capital) to tell curl to pick the filename from the last part of the URL, e.g.

```suggestion
          bat 'curl -O "https://rstudio-buildtools.s3.amazonaws.com/posit-dev/smtools-windows-x64.msi"' 
```